docs/guide/CSS.md

```diff
@@ -0,0 +1,395 @@
+
+```sass
+--8<-- "docs/examples/guide/dom4.css"
+```
+
+The CSS contains a number of rules sets with a selector and a list of rules. You can also add comments with text between `/*` and `*/` which will be ignored by Textual. Add comments to leave yourself reminders or to temporarily disable selectors.
+
+With the CSS in place, the output looks very different:
+
+```{.textual path="docs/examples/guide/dom4.py"}
+
+```
+
+### Why CSS?
+
+It is reasonable to ask why use CSS at all? Python is a powerful and expressive language. Wouldn't it be easier to do everything in your `.py` files?
+
+A major advantage of CSS is that it separates how your app _looks_ from how it _works_. Setting styles in Python can generate a lot of spaghetti code which can make it hard to see the important logic in your application.
+
+A second advantage of CSS is that you can customize builtin and third-part widgets just as easily as you can your own app or widgets.
+
+Finally, Textual CSS allows you to _live edit_ the styles in your app. If you run your application with the following command, any changes you make to the CSS file will be instantly updated in the terminal:
+
+```bash
+textual run my_app.py --dev
+```
+
+Being able to iterate on the design without restarting the Python code can make it easier and faster to design beautiful interfaces.
+
+## Selectors
+
+A selector is the text which precedes the curly braces in a set of rules. It tells textual which widgets it should apply the rules to.
+
+Selectors can target a kind of widget or a very specific widget. For instance you could have a selector that modifies all buttons, or you could target an individual button used in one dialog. This gives you a lot of flexibility in customization your user interface.
+
+Let's look at the selectors supported by Textual CSS.
+
+### Type selector
+
+The _type_ selector matches the name of the (Python) class, which is literally the name of the class in your Python code. For example, the following widget can be matched with a `Button` selector:
+
+```python
+from textual.widgets import Widget
+
+class Button(Static):
+    pass
+```
+
+The following rule applies a border to this widget:
+
+```sass
+Button {
+  border: solid blue;
+}
+```
+
+The type selector will also match a widget's base classes. Consequently a `Static` selector will also style the button because the `Button` Python class extends `Static`.
+
+```sass
+Static {
+  background: blue;
+  border: rounded white;
+}
+```
+
+!!! note
+
+    The fact that the type selector matches base classes is a departure from browser CSS which doesn't have the same concept.
+
+You may have noticed that the `border` rule exists in both Static and Button. When this happens, Textual will use the most recently defined sub-class within a list of bases. So Button wins over Static, and Static wins over Widget (the base class of all widgets). Hence if both rules were in a stylesheet, the buttons would be "solid blue" and not "rounded white".
```

Comment:
May be useful to draw an explicit parallel to how method resolution works. If `Button` and `Static` both define the method `foo`, calling `Button().foo()` actually runs `Button.foo` and not `Static.foo`.

Reply:
I'm sure there are nuances, I'm just suggesting to make an analogy here because the two things look similar.